code/linalg/linalg.c

```diff
@@ -157,10 +174,27 @@ static mp_obj_t linalg_det(mp_obj_t oin) {
         tmp[i] = ndarray_get_float_value(in->array->items, in->array->typecode, i);
     }
     mp_float_t c;
+    mp_float_t det_sign = 1.0;
```

Comment:
On some platforms, `mp_float_t det_sign = MICROPY_FLOAT_CONST(1.0)` is required. Could you, please, replace it?

Reply:
Will replace this one. I didn't know such Marco. After quick search it seems multiple files need to be changed as well?

```
$ grep -Er ' [0-9]+\.' .
./linalg/linalg.c:    mp_float_t elem = 1.0;
./linalg/linalg.c:	mp_float_t sum = 0.0;
./linalg/linalg.c:    mp_float_t det_sign = 1.0;
./linalg/linalg.c:        mp_float_t dot = 0.0;
./linalg/linalg.c:                mp_float_t sum = 0.0, v1, v2;
./linalg/linalg.c:        eigvectors[m*(in->n+1)] = 1.0;
./linalg/linalg.c:        largest = 0.0;
./linalg/linalg.c:    mp_float_t dot = 0.0, v;
./linalg/linalg.c:	mp_float_t trace = 0.0;
./ulab_create.c://|       Difference between consecutive elements, optional, defaults to 1.0
./ulab_create.c:        start = 0.0;
./ulab_create.c:        step = 1.0;
./ulab_create.c:        step = 1.0;
./ulab_create.c://|        and the other elements set to 0."""
./ulab_create.c://|    Return a new array of the given shape with all elements set to 1."""
./ulab_create.c://|    Return a new array of the given shape with all elements set to 0."""
./numerical/numerical.c:    mp_float_t value, sum = 0.0, sq_sum = 0.0;
./numerical/numerical.c:        sum = sq_sum = 0.0;
./numerical/numerical.c:        sum = 0.0;
./numerical/numerical.c:        sum_sq = 0.0;
./numerical/numerical.c:    // TODO: transpose the matrix, if axis == 0. Though, that is hard on the RAM...
./approx/approx.c://|     xtol: float = 2.4e-7,
./approx/approx.c://|     xatol: float = 2.4e-7,
./approx/approx.c://|     fatol: float = 2.4e-7,
./approx/approx.c:    //mp_float_t lambda = 0.0078125;
./approx/approx.c://|     xtol: float = 2.4e-7,
./approx/approx.c://|     rtol: float = 0.0,
./approx/approx.c://| def trapz(y: ulab.array, x: Optional[ulab.array] = None, dx: float = 1.0) -> float:
./ulab.h:// poly requires approx. 2.5 kB
./ulab.h:// the approx module consumes about 4.5 kB of flash space
./poly/poly.c:        XT[i+0*lenx] = 1.0; // top row
./poly/poly.c:            sum = 0.0;
./poly/poly.c:        sum = 0.0;
./poly/poly.c:        sum = 0.0;

```

Reply:
Also, is it necessary to change ```mp_obj_new_float(0.0);```? I guess so?

Reply:
Yeah, that's correct.

Reply:
MICROPY_FLOAT_CONST added. At least linalg.c is good to go.

Reply:
Thanks again for seeing this through till the end! You got your brownie points: https://github.com/v923z/micropython-ulab/graphs/contributors.